lib/manageiq/api/common/rbac/roles.rb

```diff
@@ -0,0 +1,77 @@
+module ManageIQ
+  module API
+    module Common
+      module RBAC
+        class Roles
+          attr_reader :roles
+
+          def initialize(prefix = nil, scope = 'principal')
+            @roles = {}
+            load(prefix, scope)
+          end
+
+          def find(name)
+            uuid = @roles[name]
+            get(uuid) if uuid
+          end
+
+          def with_each_role
+            @roles.each_value do |uuid|
+              yield get(uuid)
+            end
+          end
+
+          def add(name, acls)
+            Service.call(RBACApiClient::RoleApi) do |api_instance|
+              role_in = RBACApiClient::RoleIn.new
+              role_in.name = name
+              role_in.access = acls
+              api_instance.create_roles(role_in).tap do |role|
+                @roles[name] = role.uuid
+              end
+            end
+          end
+
+          def update(role)
+            Service.call(RBACApiClient::RoleApi) do |api_instance|
```

Comment:
Should this method be modifying `@roles` somewhere?

Reply:
We only keep name and uuid in `@roles`